### ISSUE TITLE
app.py

### ISSUE THREAD

**Issue by Rogith-007** [Rogith-007/Stock-prediction-App#1]
from flask import Flask, request, jsonify
import numpy as np
import yfinance as yf
from tensorflow.keras.models import load_model
from sklearn.preprocessing import MinMaxScaler

app = Flask(_name_)

# Load trained model
model = load_model("stock_prediction_model.h5")

# Load scaler (create and fit it again)
scaler = MinMaxScaler(feature_range=(0, 1))

@app.route('/predict', methods=['GET'])
def predict():
    stock_symbol = request.args.get('stock', default="TCS.NS", type=str)

    # Fetch last 60 days of data
    df = yf.download(stock_symbol, period="60d")[['Open', 'High', 'Low', 'Close', 'Volume']]
    
    if df.shape[0] < 60:
        return jsonify({"error": "Not enough data for prediction"})

    # Normalize data
    scaled_data = scaler.fit_transform(df)

    # Reshape for model input
    input_data = np.array([scaled_data])
    
    # Predict next day's price
    predicted_price = model.predict(input_data)
    predicted_price = scaler.inverse_transform(np.hstack((scaled_data[-1, :-1], predicted_price[0].reshape(-1, 1))))[:, -1]

    return jsonify({"Predicted Price": f"₹{predicted_price[0]:.2f}"})

if _name_ == '_main_':
    app.run(debug=True)
